### PR TITLE
Water mask refactor

### PR DESCRIPTION
**Pull request by asjohnston-asf**

### CODE REVIEW

File: hyp3_gamma/water_mask.py [hunk 1]
```diff
@@ -23,6 +23,7 @@ def create_water_mask(input_tif: str, output_tif: str):
     dst_ds = gdal.GetDriverByName('GTiff').Create(output_tif, src_ds.RasterXSize, src_ds.RasterYSize, 1, gdal.GDT_Byte)
     dst_ds.SetGeoTransform(src_ds.GetGeoTransform())
     dst_ds.SetProjection(src_ds.GetProjection())
+    dst_ds.SetMetadata(src_ds.GetMetadata())
```
Comment (asjohnston-asf) [inline hyp3_gamma/water_mask.py:26]: Setting *all* the metadata based on the input tif seems unecessary. Can we just set the `AREA_OR_POINT` field based on the source value?

Comment (asjohnston-asf) [reply hyp3_gamma/water_mask.py:26]: This might work?
```
dst_ds.SetMetadataItem('AREA_OR_POINT', src_ds.GetMetadataItem('AREA_OR_POINT')
```

Comment (cirrusasf) [reply hyp3_gamma/water_mask.py:26]: Yes. We can.

Comment (cirrusasf) [reply hyp3_gamma/water_mask.py:26]: I will change this and test it.

**Review by cirrusasf — APPROVED** [review]

**Pull request merged**